lib/thrift/binary/framed/client.ex

```diff
@@ -286,6 +308,13 @@ defmodule Thrift.Binary.Framed.Client do
       :ok ->
         {:noreply, s}
 
+      {:error, :closed} = error ->
+        if reconnect do
+          {:connect, {:reconnect, :cast, msg}, %{s | sock: nil}}
```

Comment:
We should always go through the disconnect -> connect cycle to ensure we cleanup and dont miss anything in disconnect/2. Its a little awkward but means we need to pass the reconnect info through disconnect -> connect.

Reply:
resolved in https://github.com/pinterest/elixir-thrift/pull/478/commits/6f97f5614fcaf28c03c094366f01b830a135168d